torchao/prototype/low_bit_optim/subclass_4bit.py

```diff
@@ -60,8 +61,9 @@ def __tensor_unflatten__(cls, tensor_data_dict, tensor_attributes, outer_size=No
     def dequantize(self, output_dtype=None):
         codes = torch.stack([self.codes >> 4, self.codes & 0b1111], dim=-1)  # unpack
         float_data = dequant_with_qmap(codes, self.qmap, self.scale)
-        dtype = output_dtype or torch.get_default_dtype()
-        return float_data.view(self._shape).to(dtype)
+        if output_dtype is not None:
+            float_data = float_data.to(output_dtype)
+        return float_data.view(self._shape)
```

Comment:
will we encounter non-contiguous tensor? if so, view cannot be used, reshape must be

Reply:
No, unlikely we will have non-contiguous tensor here, because this handles internal data of the tensor subclass, not outputs of another layer in a model.